src/main/java/com/thevoxelbox/voxelsniper/util/BlockHelper.java

```diff
@@ -63,4 +71,42 @@ public static boolean isSolid(BlockState state) {
         return false;
     }
 
+    public static boolean hasTraits(BlockState state, Map<BlockTrait<?>, Object> traits) {
+        for (Map.Entry<BlockTrait<?>, ?> trait : state.getTraitMap().entrySet()) {
+            Object expectedValue = traits.get(trait.getKey());
+            if (!trait.getValue().equals(expectedValue)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public static Optional<BlockState> addTraits(BlockState state, Map<BlockTrait<?>, Object> traits) {
+        Optional<BlockState> newState = Optional.of(state);
+        for (Map.Entry<BlockTrait<?>, Object> traitEntry : traits.entrySet()) {
+            newState = newState.get().withTrait(traitEntry.getKey(), traitEntry.getValue());
+            if (!newState.isPresent()) {
+                return newState;
```

Comment:
Wouldn't it be better to check if the trait applied and if it didn't, continue adding the remainder of the traits?

Reply:
I agree I'll update that.